packages/core/admin/server/services/permission/permissions-manager/sanitize.js

```diff
@@ -256,13 +260,21 @@ module.exports = ({ action, ability, model }) => {
   };
 
   const getQueryFields = (fields = []) => {
+    const nonVisibleAttributes = getNonVisibleAttributes(schema);
+    const writableAttributes = getWritableAttributes(schema);
+
+    const nonVisibleWritableAttributes = intersection(nonVisibleAttributes, writableAttributes);
+
     return uniq([
       ...fields,
       ...STATIC_FIELDS,
       ...COMPONENT_FIELDS,
+      ...nonVisibleWritableAttributes,
       CREATED_AT_ATTRIBUTE,
       UPDATED_AT_ATTRIBUTE,
       PUBLISHED_AT_ATTRIBUTE,
+      CREATED_BY_ATTRIBUTE,
+      UPDATED_BY_ATTRIBUTE,
```

Comment:
minor thing, and probably something we can work in another PR.

If now we can filter & sort by non visible fields, we should be able to set those fields as the default sort 
![image](https://github.com/strapi/strapi/assets/20578351/5426e988-2bd7-4b02-9e53-fabbdde3da40)

But that probably is another discussion.

Reply:
I agree and think we should do this as part of this PR :)

Reply:
Agree, that is why I think specifically this could be a separate discussion. On the List view, users can only sort by fields that are visible.

What I think we should allow in the list view is filtering by any nonvisible & writable field. And if we decide to do that, the work done there might help with the "Default Sort" too.

Reply:
I think that this is somting to look into when you guys do V5 doing it any sooner would make 0 sense
since we don't yet know how V5 will change this

Reply:
We have plans to refactor this for V5, but we are pushing it because we also need it in V4 :) For features like Fernando's one and also for Review Workflows.

Reply:
I see I would atleast wait until the V5 RFC is stable then we can write it in such a way that we can keep it. and that it would also for for the V5 use ceases or would work wtih v5 with minimal change.

writing it now to rewrite it in half a year makes no sence

Reply:
I am just saying we could explore this now, that will also help us write the RFC for v5.

Reply:
I agree with that it needs to be explored but I think we should explore it with the v5 structure and then back ported to v4 if possible.